cmd/influx/write.go

```diff
@@ -74,41 +90,102 @@ func cmdWrite(f *globalFlags, opt genericCLIOpts) *cobra.Command {
 	return cmd
 }
 
+func (writeFlags *writeFlagsType) dump(args []string) {
+	if writeFlags.Debug {
+		writeFlags.args = args
+		log.Printf("WriteFlags%+v", *writeFlags)
+	}
+}
+
 // createLineReader uses writeFlags and cli arguments to create a reader that produces line protocol
-func (writeFlags *writeFlagsType) createLineReader(args []string) (r io.Reader, closer io.Closer, err error) {
-	if len(args) > 0 && args[0][0] == '@' {
+func (writeFlags *writeFlagsType) createLineReader(cmd *cobra.Command, args []string) (io.Reader, io.Closer, error) {
+	readers := make([]io.Reader, 0, 2*len(writeFlags.Headers)+2*len(writeFlags.Files)+1)
+	closers := make([]io.Closer, 0, len(writeFlags.Files))
+
+	if len(args) > 0 && len(args[0]) > 1 && args[0][0] == '@' {
 		// backward compatibility: @ in arg denotes a file
-		writeFlags.File = args[0][1:]
+		writeFlags.Files = append(writeFlags.Files, args[0][1:])
+		args = args[:0]
 	}
 
-	if len(writeFlags.File) > 0 {
-		f, err := os.Open(writeFlags.File)
-		if err != nil {
-			return nil, nil, fmt.Errorf("failed to open %q: %v", writeFlags.File, err)
+	// validate input format
+	if len(writeFlags.Format) > 0 && writeFlags.Format != inputFormatLineProtocol && writeFlags.Format != inputFormatCsv {
+		return nil, write.MultiCloser(closers...), fmt.Errorf("unsupported input format: %s", writeFlags.Format)
+	}
+
+	// validate and setup decoding of files/stdin if encoding is supplied
+	decode, err := write.CreateDecoder(writeFlags.Encoding)
+	if err != nil {
+		return nil, write.MultiCloser(closers...), err
+	}
+
+	// prepend header lines
+	if len(writeFlags.Headers) > 0 {
+		for _, header := range writeFlags.Headers {
+			readers = append(readers, strings.NewReader(header), strings.NewReader("\n"))
 		}
-		closer = f
-		r = f
-		if len(writeFlags.Format) == 0 && strings.HasSuffix(writeFlags.File, ".csv") {
+		if len(writeFlags.Format) == 0 {
 			writeFlags.Format = inputFormatCsv
 		}
-	} else if len(args) == 0 || args[0] == "-" {
-		// backward compatibility: "-" also means stdin
-		r = os.Stdin
-	} else {
-		r = strings.NewReader(args[0])
 	}
-	// validate input format
-	if len(writeFlags.Format) > 0 && writeFlags.Format != inputFormatLineProtocol && writeFlags.Format != inputFormatCsv {
-		return nil, nil, fmt.Errorf("unsupported input format: %s", writeFlags.Format)
+
+	// add files
+	if len(writeFlags.Files) > 0 {
+		for _, file := range writeFlags.Files {
+			f, err := os.Open(file)
+			if err != nil {
+				return nil, write.MultiCloser(closers...), fmt.Errorf("failed to open %q: %v", file, err)
+			}
+			closers = append(closers, f)
+			readers = append(readers, decode(f), strings.NewReader("\n"))
+			if len(writeFlags.Format) == 0 && strings.HasSuffix(file, ".csv") {
+				writeFlags.Format = inputFormatCsv
+			}
+		}
+	}
+
+	// add stdin or a single argument
+	switch {
+	case len(args) == 0:
```

Comment:
there is a way to identify if stdin was provided, there is no reason we can't use both file and stdin, nothing stopping us from doing that technically.

check it out here:

https://github.com/influxdata/influxdb/blob/master/cmd/influx/pkg.go#L536

Reply:
I can change it that way too.